File: ProjectStart/ProjectStart.ViewModel/BaseViewModel.cs
using Newtonsoft.Json;
using ProjectStart.Entity.Common;
using System;
using System.Collections.Generic;
using System.Text;

namespace ProjectStart.ViewModel
{
    public class BaseViewModel<T> where T: IPageContent
    {
        public BaseViewModel(T currentPage)
        {
            CurrentPage = CurrentPage;
            SetBaseMetaData(currentPage);
        }
        public T CurrentPage { get; set; }
        public IDictionary<string, string> MetaTags { get; set; } = new Dictionary<string, string>();


        public void SetBaseMetaData(IPageContent metaTags)
        {
            MetaTags.Add("title", metaTags.Title);
            MetaTags.Add("description", metaTags.Description);
            MetaTags.Add("keyword", metaTags.Keywords);
        }
    }
}

File: ProjectStart/ProjectStart.WebApp/ClientApp/Shared/AppModels/BaseViewModel.ts
import { IPageContent } from 'shared/AppModels/IPageContent';

export interface IBaseViewModel<T>  {
    currentPage: T;
    metaTags: { [key: string]: string; };
}
export class BaseViewModel<T>  implements IBaseViewModel<T> {
    currentPage: T;
    metaTags: { [key: string]: string; };
}
